File: pset6/credit/credit.py
from cs50 import get_int

num=get_int("Number: ")
a=num
nod=0

while(a>0):
    nod+=1
    a=a//10

flag=0
total=0
a=num

for i in range(nod):
    if flag==0:
        total+=a%10
        a//=10
        flag=1
    else:
        x=2*(a%10)
        if x//10==0:
            total+=x
        else:
            total+=x%10
            x//=10
            total+=x
        a//=10
        flag=0

a=num

if total%10==0:
    for i in range(nod-2):
        a//=10
    if nod==15 and (a==34 or a==37):
        print("AMEX")
    elif nod==16 and (a>=51 and a<=55):
        print("MASTERCARD")
    elif (nod==16 or nod==13) and a//10==4:
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")

File: pset1/credit.c
#include<cs50.h>
#include<stdio.h>

int main(void)
{
    int flag=1,nod=1; //nod stands for number of digits
    long int num=get_long("Number: "); //input credit card number
    long int a=num;
    while(flag) //find the number of digits in the number
    {
        if(a/10!=0)
        {
            nod++;
            a/=10;
        }
        else
            flag=0;
    }
    a=num;
    int sum=0,x; 
    for(int i=nod; i>0; i--) //checksum is being applied
    {
        if(flag==0)
        {
            sum+=a%10;
            a/=10;
            flag=1;
        }
        else
        {
            x=2*(a%10);
            if(x/10==0)
                sum+=x;
            else
            {
                sum+=(x%10);
                x/=10;
                sum+=x;
            }
            a/=10;
            flag=0;
        }
    }
    a=num;
    if(sum%10==0)
    {
        for(int i=nod; i>2; i--) //the variable a stores the first two digits of the number
            a/=10;
        if(nod==15 && (a==34 || a==37)) //determines the type of credit card
            printf("AMEX\n");
        else if(nod==16 && (a>=51 && a<=55))
            printf("MASTERCARD\n");
        else if((nod==16 || nod==13) && a/10==4)
            printf("VISA\n");
        else
            printf("INVALID\n");
    }
    else
        printf("INVALID\n");
}